Add deprecation warning about QR codes (#937)
shlinkio/shlink.io: src/pages/documentation/environment-variables.mdx

#### QR codes

<Callout type="warning">

>   Generating QR Codes through Shlink is being deprecated (see <a href="https://github.com/shlinkio/shlink/issues/2408">#2408</a>).

Reviewer: I reckon markdown works inside JSX when used in MDX files.

Also, I think it's better to be more specific on where the link is taking you, as an issue number in isolation does not have a lot of meaning in this context.  

```suggestion
  Generating QR codes through Shlink is being deprecated. See [this issue](https://github.com/shlinkio/shlink/issues/2408) for details.
```
